internal/pkg/service/buffer/storage/local/writer/disksync/disksync.go

```diff
@@ -117,168 +117,141 @@ func NewSyncer(logger log.Logger, clock clock.Clock, config Config, chain chain)
-		return nil
+// Stop periodical synchronization.
+func (s *Syncer) Stop() error {
+	if err := s.ctx.Err(); err != nil {
+		return errors.Errorf(`syncer is already stopped: %w`, err)
 	}
 
-	s.notifierLock.RLock()
-	notifier := s.notifier
-	do()
-	s.notifierLock.RUnlock()
+	s.logger.Debug(`stopping syncer`)
 
-	return notifier
-}
+	// Stop sync loop
+	s.cancel()
+
+	// Run last sync
+	_ = s.TriggerSync(true).Wait()
 
-// SyncAndWait initiates synchronization and waits for its completion.
-// If a synchronization is already in progress, the method waits for it to complete before starting a new one.
-// Method is public for tests.
-func (s *Syncer) SyncAndWait() {
-	// Wait, if the sync is already in progress
-	s.syncLock.Lock()
+	// Wait for sync loop and running sync, if any
+	s.wg.Wait()
 
-	// Wait for sync completion
-	_ = s.doSync().Wait()
+	s.logger.Debug(`syncer stopped`)
+	return nil
 }
 
-// Sync initiates synchronization if it is not already running.
-// Method is public for tests.
-func (s *Syncer) Sync() {
-	// Skip, if the sync is already in progress
-	if !s.syncLock.TryLock() {
-		return
+// TriggerSync initiates synchronization.
+// If force=true, it waits for a running synchronization, if there is one, and then starts a new one.
+// If force=false, is doesn't wait, a notifier for the running synchronization returns.
+// In both cases, the method doesn't wait for the synchronization to complete,
+// you can use the Wait() method of the returned *notify.Notifier for waiting.
+func (s *Syncer) TriggerSync(force bool) *notify.Notifier {
+	// Check if the sync is disabled
+	if s.opFn == nil {
+		return nil
 	}
 
-	// Run sync in the background
-	s.doSync()
-}
-
-func (s *Syncer) onWrite(l int) {
-	// Increment counter and check size condition
-	if datasize.ByteSize(s.bytesToSync.Add(uint64(l))) >= s.config.BytesTrigger {
-		s.bytesTriggerCh <- struct{}{}
+	// Acquire the syncLock
+	if force {
+		s.syncLock.Lock()
+	} else if !s.syncLock.TryLock() {
+		// Skip trigger, if a sync is already in progress
+		s.notifierLock.RLock()
+		defer s.notifierLock.RUnlock()
+		return s.notifier
 	}
-}
 
-func (s *Syncer) startSyncLoop() {
-	s.wg.Add(1)
-	go func() {
-		defer s.wg.Done()
-		defer s.timer.Stop()
+	// At this point the syncLock is locked.
+	// It is released at the end of the goroutine bellow.
 
-		for {
-			select {
-			case <-s.ctx.Done():
-				// The Close method has been called
-				return
-			case <-s.bytesTriggerCh:
-				// SyncAfterBytes
-				s.Sync()
-			case <-s.timer.C:
-				// SyncAfterInterval
-				s.Sync()
-			}
-		}
-	}()
-}
+	// Update counters
+	s.writeOpsCount.Store(0)
+	s.lastSyncAt.Store(s.clock.Now())
+	s.bytesToSync.Store(0)
 
-func (s *Syncer) doSync() *notify.Notifier {
 	// Swap sync notifier, split old and new writes
 	s.notifierLock.Lock()
 	notifier := s.notifier
 	s.notifier = notify.New()
 	s.notifierLock.Unlock()
 
-	// Schedule next periodical sync
-	s.timer.Reset(s.config.IntervalTrigger)
```

Comment:
Predtym tu bol `timer`, ktory sa vzdy resetoval, teraz tam je `ticker 5ms`.

To bol tiez problem, ak sync trval dlhsie ako timer interval, tak sa dalsi `sync` spustil pocas predchadzajuceho, `sync` trgger sa ignoroval, ale timer sa uz nikdy neresetoval.